pkg/catalog/catalog.go

```diff
@@ -598,36 +598,25 @@ func (c *Catalog) ListRepositories(ctx context.Context, limit int, prefix, after
 	defer it.Close()
 	// seek for first item
 	afterRepositoryID := graveler.RepositoryID(after)
-	prefixRepositoryID := graveler.RepositoryID(prefix)
-	startPos := prefixRepositoryID
-	if afterRepositoryID > startPos {
-		startPos = afterRepositoryID
```

Comment:
Doesn't this break pagination?

Reply:
Yes you are correct thanks. Added some failing tests and fixed it